presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java

```diff
@@ -153,6 +154,16 @@ public StatsAndCosts getStatsAndCosts()
         return jsonRepresentation;
     }
 
+    // Serialize this plan fragment with the provided codec, caching the results
+    public SerializedString toBytes(Codec<PlanFragment> codec)
+    {
+        requireNonNull(codec, "codec is null");
+        if (cachedSerialization == null) {
```

Comment:
benign data race ? 😃 

The only question is if all the serialization request comes at similar time, would this cause we still serialize for every task?

Reply:
I'll just synchronize it